resources/authgear/templates/en/web/authflowv2/__settings_radio.html

```diff
@@ -0,0 +1,37 @@
+<!-- Example usage:
+{{ template "authflowv2/__settings_radio.html"
+  (dict
+    "Label" (include "some-key" nil)
+    "RadioName" "gender-select"
+    "RadioValue" "male"
+    "ShowInput" false
+    "InputName" "gender-input"
+    "InputValue" "customGender"
+    "DefaultChecked" true
+  )
+}}
+-->
+{{ define "authflowv2/__settings_radio.html" }}
+<label class="radio-label" for={{ $.RadioValue }}>
+  {{ $.Label }}
+
+  <input
+    type="radio"
+    class="radio"
+    id={{ $.RadioValue }}
+    name="{{ $.RadioName }}"
+    value="{{ $.RadioValue }}"
+    {{ if $.DefaultChecked }}checked{{end}}
+  >
+
+  {{ if $.ShowInput }}
+    <input
+      class="input radio-input col-span-2"
+      name="{{ $.InputName }}"
+      value="{{ $.InputValue }}"
+    >
+  {{ end }}
```

Comment:
We wont always assume a radio is always associated with an input. Actually it can show anything we want.

I suggest the api should look like

```html
// radio.html

{{ define "radio" }}
    <div>
        <label>
            <input />
        </label> 
        <div class="display-only-when-checked">
            {{ OnCheckShowElement }}
        </div>
    </div>
{{ end }}

// profile_edit.html

{{ template "radio"
    (dict 
        "OnCheckShowElement" (template "__settings_profile_edit__other_gender_input")
    )
}}

{{ define "__settings_profile_edit__other_gender_input "}}
    <input type="text" name"gender-other" />
{{ end }}
```